app/src/main/java/de/westnordost/streetcomplete/overlays/street_furniture/StreetFurnitureOverlayForm.kt

```diff
@@ -0,0 +1,159 @@
+                ?: DummyFeature(
+                    "street_furniture/unknown",
+                    requireContext().getString(R.string.unknown_street_furniture),
+                    "ic_preset_maki_marker_stroked",
+                    element.tags
+                )
+            }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // title hint label with name is a duplication, it is displayed in the UI already
+        setTitleHintLabel(element?.tags?.let { getLocationLabel(it, resources) })
+        setMarkerIcon(R.drawable.ic_quest_apple)
+
+        featureCtrl = FeatureViewController(featureDictionary, binding.featureTextView, binding.featureIconView)
+        featureCtrl.countryOrSubdivisionCode = countryOrSubdivisionCode
+        featureCtrl.feature = originalFeature
+
+        binding.featureView.setOnClickListener {
+            SearchFeaturesDialog(
+                requireContext(),
+                featureDictionary,
+                element?.geometryType,
+                countryOrSubdivisionCode,
+                featureCtrl.feature?.name,
+                ::filterOnlyStreetFurniture,
+                ::onSelectedFeature,
+                listOf(
+                    // ordered by popularity, skipping trees as there are multiple variants of them
+                    "highway/street_lamp",
+                    "amenity/bench",
+                    "emergency/fire_hydrant",
+                    "amenity/bicycle_parking",
+                    "amenity/shelter",
+                    "amenity/toilets",
+                    // "amenity/post_box",
+                    // blocked by https://github.com/streetcomplete/StreetComplete/issues/4916
+                    // waiting for response in https://github.com/ideditor/schema-builder/issues/94
+                    "amenity/drinking_water",
+                    "leisure/picnic_table",
+
+                    // popular, a bit less than some competing entries
+                    // but interesting and worth promoting
+                    "emergency/defibrillator",
+                    )
+            ).show()
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+    }
+
+    private fun filterOnlyStreetFurniture(feature: Feature): Boolean {
+        val fakeElement = Node(-1L, LatLon(0.0, 0.0), feature.tags, 0)
+        return IS_STREET_FURNITURE_INCLUDING_DISUSED_EXPRESSION.matches(fakeElement)
+    }
+
+    private fun onSelectedFeature(feature: Feature) {
+        featureCtrl.feature = feature
+        checkIsFormComplete()
+    }
+
+    override fun hasChanges(): Boolean =
+        originalFeature != featureCtrl.feature
+
+    override fun isFormComplete(): Boolean = featureCtrl.feature != null
+
+    override fun onClickOk() {
+        viewLifecycleScope.launch {
```

Comment:
Why this `launch` and why is `createEditAction` a suspend function?

Reply:
Yes, that is leftover from shop overlay (this one started as a direct copy of it,)